tests/utils/test_utils.py

```diff
@@ -119,24 +119,47 @@ def compare_categories(test, expected, actual):
 IGNORE_ALL = "*"
 
 
-def _compare_annotations(expected, actual, ignored_attrs=None):
-    if not ignored_attrs:
-        return expected == actual
+def _compare_annotations(
+    expected,
+    actual,
+    ignored_attrs=None,
+    ignore_ann_id: bool = False,
+    ignore_ann_group: bool = False,
+):
+    a_id, b_id = getattr(expected, "id"), getattr(actual, "id")
+    a_group, b_group = getattr(expected, "group"), getattr(actual, "group")
+    a_attr, b_attr = expected.attributes, actual.attributes
 
-    a_attr = expected.attributes
-    b_attr = actual.attributes
+    if ignore_ann_id:
+        setattr(expected, "id", 0)
+        setattr(actual, "id", 0)
 
-    if ignored_attrs != IGNORE_ALL:
-        expected.attributes = filter_dict(a_attr, exclude_keys=ignored_attrs)
-        actual.attributes = filter_dict(b_attr, exclude_keys=ignored_attrs)
-    else:
+    if ignore_ann_group:
+        setattr(expected, "group", 0)
+        setattr(actual, "group", 0)
+
+    if ignored_attrs == None:
```

Comment:
```suggestion
    if ignored_attrs is None:
```
from pep8 style guide,
Comparisons to singletons like None should always be done with is or is not, never the equality operators.